framework/TApplication.php

```diff
@@ -1256,7 +1256,7 @@ public function flushOutput($continueBuffering = true)
 	public function onEndRequest()
 	{
 		$this->flushOutput(false); // flush all remaining content in the buffer
-		$this->saveGlobals();  // save global state
 		$this->raiseEvent('OnEndRequest', $this, null);
+		$this->saveGlobals();  // save global state
```

Comment:
Please note that the globalstate has already been saved on the onSaveState step.
This call to saveGlobals() is only a last-time fallback to ensure globals are saved, but changing the global sta ein the flush/OnEndRequest step will trigger a second forced saving with all the relative performance loss.